### PR TITLE
feat: support for dot (.) in attribute names

### PR DESCRIPTION
**Pull request by nityanandagohain**
Fixes https://github.com/SigNoz/signoz/issues/3553

This PR adds the following changes.

If there is an attribute with dot in name ex:- `service.name`
* Two attributes will be stored with the log line `service_name` and `service.name`
* `service_name` won't be added to tag_attributes as we don't want users to use the underscore attribute name in new queries. But the exisiting queries will work as we are storing the `_` attribute with the log line.

After 20 days / 1 month remove the `addTemporaryUnderscoreSupport` function from the code.

### CODE REVIEW

**Comment by srikanthccv** [comment]
We don't need to do this for new users. If it's easier to make this double write run only for old users then let's do that instead.

**Comment by nityanandagohain** [comment]
I think keeping this will be better than managing the config for all the users.

As the `_` attributes won't show up anywhere and user won't be able to materialize as well. just that for sometime the attributes column of a log line will be double.

Let me know your thoughts

**Comment by srikanthccv** [comment]
I am fine with it. If there was a _easier_ way to enable this to old users only we could avoid unnecessary double storage for new users.

**Comment by nityanandagohain** [comment]
Yeah, let me check if something can be done with @prashant-shahi  once. If we have to update all deployment files I will keep this as it is.

**Review by srikanthccv — APPROVED** [review]

File: exporter/clickhouselogsexporter/exporter.go [hunk 1]
```diff
@@ -338,31 +346,64 @@ func addAttrsToTagStatement(statement driver.Batch, tagType string, attrs attrib
 
 func attributesToSlice(attributes pcommon.Map, forceStringValues bool) (response attributesToSliceResponse) {
 	attributes.Range(func(k string, v pcommon.Value) bool {
+		// Support for . , remove the above section once done.
```
[the remark this answers is not in the document]
Comment (nityanandagohain) [reply exporter/clickhouselogsexporter/exporter.go:349]: oh I meant the addTemporaryUnderscoreSuppor

**Pull request merged**